pytest_plugins/testimony_markers.py

```diff
@@ -0,0 +1,106 @@
+import inspect
+import logging
+import re
+
+import pytest
+
+LOGGER = logging.getLogger('robottelo')
+
+IMPORTANCE_LEVELS = []
+
+
+def pytest_addoption(parser):
+    """Add CLI options related to Testimony token based mark collection"""
+    parser.addoption(
+        '--importance',
+        help='Comma separated list of importance levels to include in collection',
+    )
+    parser.addoption(
+        '--component',
+        help="Comma separated list of component names to include in collection",
+    )
+
+
+def pytest_configure(config):
+    """Register markers related to testimony tokens"""
+    for marker in [
+        'importance: CaseImportance testimony token, use --importance to filter',
+        'component: Component testimony token, use --component to filter',
+        # TODO: components read from testimony.yaml
+    ]:
+        config.addinivalue_line("markers", marker)
+
+
+component_regex = re.compile(
+    # To match :CaseComponent: FooBar
+    r"\s*:CaseComponent:\s*(?P<component>\S*)",
+    re.IGNORECASE,
+)
+
+importance_regex = re.compile(
+    # To match :CaseImportance: Critical
+    r"\s*:CaseImportance:\s*(?P<importance>\S*)",
+    re.IGNORECASE,
+)
+
+
+@pytest.hookimpl(tryfirst=True)
```

Comment:
We have two plugins with tryfirst (rerun_rp) , which one will be used first ? As per the sequence in conftest list ?

Reply:
There's no guarantee within pytest what order the hooks will be executed in, but these two hooks should be fine regardless of the order that they run in. There will be more `items` loop iterations in whichever runs first, and fewer `items` loop iterations in whichever runs second.